File: VB/PictureAndLinq/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Collections

Namespace PictureAndLinq
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			repositoryItemPictureEdit1.PictureStoreMode = DevExpress.XtraEditors.Controls.PictureStoreMode.ByteArray

			Dim dc As New AdventureWorksDataContext()
			Dim bs As New BindingSource(dc.ProductPhotos, "")
			gridControl1.DataSource = bs
		End Sub

		Private Sub gridView1_CustomUnboundColumnData(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs) Handles gridView1.CustomUnboundColumnData
			If e.Column.FieldName = "LargePhoto_Unbound" Then
				Dim view As GridView = CType(sender, GridView)
				Dim dataSource As IList = CType(view.DataSource, IList)
				Dim pf As ProductPhoto = CType(dataSource(e.ListSourceRowIndex), ProductPhoto)
				If pf Is Nothing Then
					Return
				End If
				If e.IsGetData Then
					e.Value = pf.LargePhoto.ToArray()
				ElseIf e.IsSetData Then
					Dim data() As Byte = CType(e.Value, Byte())
					pf.LargePhoto = New System.Data.Linq.Binary(data)
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/PictureAndLinq/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Views.Grid;
using System.Collections;

namespace PictureAndLinq {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            repositoryItemPictureEdit1.PictureStoreMode = DevExpress.XtraEditors.Controls.PictureStoreMode.ByteArray;

            AdventureWorksDataContext dc = new AdventureWorksDataContext();
            BindingSource bs = new BindingSource(dc.ProductPhotos, "");
            gridControl1.DataSource = bs;
        }

        private void gridView1_CustomUnboundColumnData(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs e) {
            if(e.Column.FieldName == "LargePhoto_Unbound") {
                GridView view = (GridView)sender;
                IList dataSource = (IList)view.DataSource;
                ProductPhoto pf = (ProductPhoto)dataSource[e.ListSourceRowIndex];
                if(pf == null) return;
                if(e.IsGetData) {
                    e.Value = pf.LargePhoto.ToArray();
                }
                else if(e.IsSetData) {
                    byte[] data = (byte[])e.Value;
                    pf.LargePhoto = new System.Data.Linq.Binary(data);
                }
            }
        }
    }
}
